.github/workflows/fleximod_test.yaml

```diff
@@ -0,0 +1,31 @@
+name: git-fleximod test
+#
+# Test git-fleximod update and cleanliness
+# Based closely on workflow from CESM repo
+#
+on: [push, pull_request]
+
+jobs:
+  fleximod-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # oldest supported and latest supported
+        python-version: ["3.7", "3.x"]
+    steps:
+      - id: checkout-CTSM
+        uses: actions/checkout@v4
+      - id: run-fleximod
+        run: |
+          $GITHUB_WORKSPACE/bin/git-fleximod update
+          echo
+          echo "Update complete, checking status"
+          echo
+          $GITHUB_WORKSPACE/bin/git-fleximod test
+      - id: check-cleanliness
+        run: |
+          echo
+          echo "Checking if git fleximod matches expected externals"
```

Comment:
Ahh, nice. So this will make sure that the submodule directory was updated in addition to the .gitmodule file.